app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java

```diff
@@ -2,16 +2,74 @@
 
 import com.example.mycalculator.mvp.contract.CalculatorContract;
 
+import static com.example.mycalculator.symbols.OperatorsSymbols.operatorDivide;
+import static com.example.mycalculator.symbols.OperatorsSymbols.operatorMinus;
+import static com.example.mycalculator.symbols.OperatorsSymbols.operatorMultiply;
+import static com.example.mycalculator.symbols.OperatorsSymbols.operatorSum;
+
+import static com.example.mycalculator.utils.StringUtils.EMPTY_STRING;
+
 public class CalculatorModel implements CalculatorContract.CalculatorModelContract {
 
-    private String firstOperand;
+    private String firstOperand = EMPTY_STRING;
+    private String secondOperand = EMPTY_STRING;
+    private String operator = EMPTY_STRING;
+    private String result = EMPTY_STRING;
+
 
     @Override
     public void saveNumber(String number) {
-        firstOperand = number;
+        if (operator.isEmpty()) {
+            firstOperand += number;
+        } else {
+            secondOperand += number;
+        }
+    }
+
+    public String getPartialResult() {
+
+        return (firstOperand + operator + secondOperand);
+    }
+
+    @Override
+    public String getFullResult() {
+
+        switch (operator) {
+
+            case operatorSum:
+                result = String.valueOf((Double.parseDouble(firstOperand) + Double.parseDouble(secondOperand)));
+                break;
+
+            case operatorMinus:
+                result = String.valueOf((Double.parseDouble(firstOperand) - Double.parseDouble(secondOperand)));
+                break;
+
+            case operatorDivide:
+                result = String.valueOf((Double.parseDouble(firstOperand) / Double.parseDouble(secondOperand)));
+                break;
+
+            case operatorMultiply:
+                result = String.valueOf((Double.parseDouble(firstOperand) * Double.parseDouble(secondOperand)));
+                break;
+
+            default:
+                result = "Error, unknow Symbol used";
```

Comment:
no hardcode pls, create a constant or return a specific value so in the presenter you can identify it and cal a showErrorMessage in the view

Reply:
Done. Thx